server/src/main/java/org/elasticsearch/action/update/UpdateHelper.java

```diff
@@ -258,7 +268,8 @@ Result prepareUpdateScriptRequest(ShardId shardId, UpdateRequest request, GetRes
                     .setIfPrimaryTerm(getResult.getPrimaryTerm())
                     .waitForActiveShards(request.waitForActiveShards())
                     .timeout(request.timeout())
-                    .setRefreshPolicy(request.getRefreshPolicy());
+                    .setRefreshPolicy(request.getRefreshPolicy())
+                    .setNormalisedBytesParsed(0L);// an indicator that this request should not be reported
```

Comment:
Related: currently the logic to interpret the content of `IndexRequest#normalizedBytesParsed` is inside `TransportShardBulkAction#getDocumentSizeObserver`. The good thing is that it is one place; the bad thing is that it is separated.
Maybe you can transform the getter to a visitor (e.g. `IndexRequest#visitNormalisedBytesParsed(visitor)`, where the visitor calls different methods based on the state - so the logic is inside IndexRequest.
Or if you feel this is a bit too complicated, maybe you can have boolean methods on IndexRequest (e.g. `isDocumentAlreadyParsed()` or `hasNothingToReport()` where you have the checks against the values (`normalizedBytesParsed != -1`, `normalizedBytesParsed == 0`)

Just an idea that may improve a bit, but I think it is OK the way it is too.

Reply:
I like `noParsedBytesToReport` method name. This will allow to remove the comment.

regarding the TransportShardBulkAction#getDocumentSizeObserver I still feel it would be better to have this method in serverless. We could perhaps always pass down an IndexRequest or a long to getDocumentSizeObserver
I will leave this as a follow up for the future